File: codegen/results/v002/qwen3.5_27b_nothink/20260411_194638/python/008_prime_numbers/solution.py
import sys

def is_prime(n: int) -> bool:
    """Check if a number is prime."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    # Check odd divisors up to sqrt(n)
    limit = int(n ** 0.5) + 1
    for i in range(3, limit, 2):
        if n % i == 0:
            return False
    return True

def main():
    if len(sys.argv) != 2:
        # If no argument provided, exit silently or handle as needed.
        # The problem implies exactly one argument, so we exit if invalid.
        return

    try:
        n = int(sys.argv[1])
    except ValueError:
        # If the argument is not a valid integer, exit silently.
        return

    # Print all primes up to and including n
    for num in range(2, n + 1):
        if is_prime(num):
            print(num)

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.5_27b_nothink/20260411_194638/typescript/008_prime_numbers/solution.ts
/**
 * Sieve of Eratosthenes to find all prime numbers up to N.
 * 
 * @param n The upper limit (inclusive)
 * @returns An array of prime numbers up to n
 */
function getPrimesUpTo(n: number): number[] {
  if (n < 2) {
    return [];
  }

  // Initialize a boolean array to track prime status.
  // isPrime[i] will be false if i is prime, true if composite.
  const isComposite: boolean[] = new Array(n + 1).fill(false);
  const primes: number[] = [];

  for (let i = 2; i <= n; i++) {
    if (!isComposite[i]) {
      primes.push(i);

      // Mark multiples of i starting from i*i
      if (BigInt(i) * BigInt(i) <= BigInt(n)) {
        for (let j = i * i; j <= n; j += i) {
          isComposite[j] = true;
        }
      }
    }
  }

  return primes;
}

function main() {
  // Read command line argument
  if (process.argv.length < 3) {
    // No argument provided, print nothing
    return;
  }

  const input = process.argv[2];
  const n = parseInt(input, 10);

  if (isNaN(n)) {
    // If input is not a valid integer, print nothing or handle error
    // The problem implies N is an integer, so we assume valid integer input or print nothing
    return;
  }

  const primes = getPrimesUpTo(n);

  if (primes.length > 0) {
    console.log(primes.join('\n'));
  }
}

main();